source/application/AVIDumping/AVIDumper.cpp

```diff
@@ -57,680 +57,754 @@ extern Movie movie;
-		}
-
-		if (!soundInfo.format)
-		{
-			return;
-		}
-
-		WAVEFORMATEX format;
-		if (!ReadProcessMemory(captureProcess, soundInfo.format, &format, sizeof(WAVEFORMATEX), NULL))
-		{
-			return;
-		}
-
-		Slot& slot = m_slots[m_nextWriteAudio];
-        VerboseDebugLog() << "FillEmptyAudioFrame: slot " << m_nextWriteAudio << ", movie frame " << movie.currentFrame;
-		m_nextWriteAudio++;
-		m_nextWriteAudio %= numSlots;
-
-		slot.hasProcessedAudio.WaitUntilFalse();
-
-		slot.audioFrameSamples = soundInfo.size / format.nBlockAlign;
-
-		ReserveBuffer(slot.audioBuffer, slot.audioBufferAllocated, soundInfo.size);
-		if (format.wBitsPerSample == 8)
-		{
-			memset(slot.audioBuffer, 0x80, soundInfo.size);
-		}
-		else
-		{
-			memset(slot.audioBuffer, 0, soundInfo.size);
-		}
-
-		slot.inAudioSize = soundInfo.size;
-		slot.inAudioSeconds = (double)soundInfo.size / format.nAvgBytesPerSec;
-
-		slot.hasProcessedAudio = true;
-	}
-
-	struct Slot
-	{
-		WaitableBool hasProcessedVideo;
-		WaitableBool hasProcessedAudio;
-		unsigned char* aviPixels;
-		int aviPixelsAllocated;
-		unsigned char* audioBuffer;
-		int audioBufferAllocated;
-		int inAudioSize, audioFrameSamples;
-		double inAudioSeconds; // hack
-		int framecount;
-		int slotNum; // for debugging
-		Slot() : hasProcessedVideo(false), hasProcessedAudio(false),
-			aviPixels(NULL), aviPixelsAllocated(0),
-			audioBuffer(NULL), audioBufferAllocated(0),
-			audioFrameSamples(0), inAudioSize(0), inAudioSeconds(0)
-		{
-			framecount = -1;
-			slotNum = 0;
-		}
-
-		~Slot()
-		{
-			Deallocate();
-		}
-
-		void Deallocate()
-		{
-			free(aviPixels);
-			aviPixels = NULL;
-			free(audioBuffer);
-			audioBuffer = NULL;
-			aviPixelsAllocated = 0;
-			audioBufferAllocated = 0;
-		}
-
-		private:
-
-			void OutputVideoFrame()
-			{
-				if (Config::localTASflags.aviMode & 1)
-				{
-					int videoFrameSize = curAviWidth * curAviHeight * (24 / 8);
-					LONG bytesWritten = 0;
-
-					{
-						AutoCritSect cs(&s_aviCS);
-
-						if (!aviCompressedStream)
-						{
-							return;
-						}
+        slot.framecount = movie.currentFrame;
+
+        const int aviPixelsSize = width * height * (24 / 8) + 8;
+        ReserveBuffer(slot.aviPixels, slot.aviPixelsAllocated, aviPixelsSize);
+
+        unsigned char* aviPix = slot.aviPixels;
+
+        const int bytesPerInPixel = bpp >> 3;
+        const int bytesPerAVIPixel = 24 >> 3;
+
+        int rShiftLeft = leftShiftFromMask(rmask);
+        int gShiftLeft = leftShiftFromMask(gmask);
+        int bShiftLeft = leftShiftFromMask(bmask);
+        const int rShiftRight = std::max(0, -rShiftLeft);
+        const int gShiftRight = std::max(0, -gShiftLeft);
+        const int bShiftRight = std::max(0, -bShiftLeft);
+        rShiftLeft = std::max(0, rShiftLeft);
+        gShiftLeft = std::max(0, gShiftLeft);
+        bShiftLeft = std::max(0, bShiftLeft);
+
+        if (bytesPerInPixel == 4 && rmask == 0x00FF0000 && gmask == 0x0000FF00
+            && bmask == 0x000000FF)
```

Comment:
Hm, maybe there's some rule to adjust this? Like "always end line with and/or statement"?

Reply:
Do you mean keeping `&&` and other operators at the end of the previous line versus on the new line? I deliberately set it like this because this way it's clear what operation is applied to the argument on the new line.

Reply:
For me it's actually clearer when it's the other way around. But ok, I think I can live with this.